src/node/interfaces.cpp

```diff
@@ -883,11 +883,24 @@ class MinerImpl : public Mining
         return TestBlockValidity(state, chainman().GetParams(), chainman().ActiveChainstate(), block, tip, /*fCheckPOW=*/false, check_merkle_root);
     }
 
-    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& script_pub_key, bool use_mempool) override
+    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& script_pub_key, bool use_mempool,
+                                                   size_t coinbase_output_max_additional_size,
+                                                   size_t coinbase_output_max_additional_sigops) override
     {
         BlockAssembler::Options options;
         ApplyArgsManOptions(gArgs, options);
 
+        Assume(coinbase_output_max_additional_size <= DEFAULT_BLOCK_MAX_WEIGHT);
```

Comment:
This probably needs to be `DEFAULT_BLOCK_MAX_WEIGHT / 4`, let me check the spec as well...